tests/e2e/prevent_csi_session_leaks.go

```diff
@@ -0,0 +1,174 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+
+	ginkgo "github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	"github.com/vmware/govmomi/property"
+	"github.com/vmware/govmomi/vim25/mo"
+	vimtypes "github.com/vmware/govmomi/vim25/types"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	fnodes "k8s.io/kubernetes/test/e2e/framework/node"
+	fpod "k8s.io/kubernetes/test/e2e/framework/pod"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+var _ = ginkgo.Describe("[csi-block-vanilla][csi-file-vanilla][csi-supervisor]"+
+	" Prevent CSI Session Leak Tests", func() {
+	f := framework.NewDefaultFramework("session-leaks-csi")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
+	var (
+		client          clientset.Interface
+		c               clientset.Interface
+		csiReplicaCount int32
+		deployment      *appsv1.Deployment
+	)
+
+	ginkgo.BeforeEach(func() {
+		bootstrap()
+		client = f.ClientSet
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		nodeList, err := fnodes.GetReadySchedulableNodes(f.ClientSet)
+		framework.ExpectNoError(err, "Unable to find ready and schedulable Node")
+		if !(len(nodeList.Items) > 0) {
+			framework.Failf("Unable to find ready and schedulable Node")
+		}
+		// Get CSI Controller's replica count from the setup
+		controllerClusterConfig := os.Getenv(contollerClusterKubeConfig)
+		c = client
+		if controllerClusterConfig != "" {
+			framework.Logf("Creating client for remote kubeconfig")
+			remoteC, err := createKubernetesClientFromConfig(controllerClusterConfig)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			c = remoteC
+		}
+		deployment, err = c.AppsV1().Deployments(csiSystemNamespace).Get(ctx,
+			vSphereCSIControllerPodNamePrefix, metav1.GetOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		csiReplicaCount = *deployment.Spec.Replicas
+	})
+
+	ginkgo.It("Verify session creted with unique user agent", func() {
+		var cancel context.CancelFunc
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		// check sessions are uniquely identifiable
+		tasksListViewEnabled := isCsiFssEnabled(ctx, client, GetAndExpectStringEnvVar(envCSINamespace),
+			"listview-tasks")
+		gomega.Expect(tasksListViewEnabled).To(gomega.BeTrue())
+		sessionList := getSessionList(ctx)
+		count := 0
+		matchRgx := "k8s-csi-useragent-[^.]+-[^.]+"
+		rgx, err := regexp.Compile(matchRgx)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		for _, session := range sessionList {
+			matched := rgx.MatchString(session.UserAgent)
```

Comment:
this should be `matched, err := rgx.MatchString(session.UserAgent)` right
please correct this in all the places
also you can move this to a separate func

Reply:
moved to new function